src/python/pants/notes/2.18.x.md

```diff
@@ -1,349 +1,8 @@
-# 2.18.x Release Series
+# 2.17.x Release Series
```

Comment:
Based on the filename, this should be `2.18.x`?

Also, the `2.18.x` branch was cut via `2.18.0a0` on August 19th-ish, so assuming that we want to use this process for that series, then this change should be cherry-picked to `2.18.x`.